### ISSUE TITLE
[docker-images] Please update `4.0.x` to use Alpine and the new images like `develop`

### ISSUE THREAD

**Issue by sumerjabri** [craftercms/craftercms#6482]
### Duplicates

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

The `support/4.0.x` branch is still on the old images.

### Describe the solution you'd like

Please update to be like `develop` and `4.1.x` and use Alpine and the new mechanics.

**Comment by Euquimides** [comment]
Hi @phuongnq ! While testing this feature, I noticed that I can build the image without problems using v4.0.8, but when I run the image through `docker-compose`, a context initialization error shows up through the logs for `authoring-tomcat`:
```
authoring-tomcat-1         | [ERROR] 2024-02-07T20:35:47,627 [main] [] [context.ContextLoader] | Context initialization failed 
authoring-tomcat-1         | org.springframework.beans.factory.CannotLoadBeanClassException: Error loading class [org.craftercms.core.cache.impl.store.guava.GuavaCacheStoreAdapter] for bean with name 'crafter.cacheStoreAdapter' defined in class path resource [crafter/core/cache-context.xml]: problem with class file or dependent class; nested exception is java.lang.UnsupportedClassVersionError: org/craftercms/core/cache/impl/store/guava/GuavaCacheStoreAdapter has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0 (unable to load class [org.craftercms.core.cache.impl.store.guava.GuavaCacheStoreAdapter])
authoring-tomcat-1         | 	at org.springframework.beans.factory.support.AbstractBeanFactory.resolveBeanClass(AbstractBeanFactory.java:1559) ~[spring-beans-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineTargetType(AbstractAutowireCapableBeanFactory.java:704) ~[spring-beans-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.predictBeanType(AbstractAutowireCapableBeanFactory.java:674) ~[spring-beans-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1684) ~[spring-beans-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1161) ~[spring-beans-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:935) ~[spring-beans-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:929) ~[spring-context-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:591) ~[spring-context-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:399) ~[spring-web-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:278) [spring-web-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103) [spring-web-5.3.31.jar:5.3.31]
authoring-tomcat-1         | 	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4462) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:4914) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:683) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:658) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:661) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:1025) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:1919) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
authoring-tomcat-1         | 	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
authoring-tomcat-1         | 	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) [tomcat-util.jar:9.0.85]
authoring-tomcat-1         | 	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:118) [?:?]
authoring-tomcat-1         | 	at org.apache.catalina.startup.HostConfig.deployWARs(HostConfig.java:826) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:476) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1619) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:318) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:114) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:402) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:345) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:893) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:794) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1332) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1322) [catalina.jar:9.0.85]
authoring-tomcat-1         | 	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
authoring-tomcat-1         | 	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) [tomcat-util.jar:9.0.85]
```
Then, when accessing Studio, a 404 page shows up: 
![image](https://github.com/craftercms/craftercms/assets/90878288/4701d8ce-7966-434d-b46f-25e82791caab)
 
Seems this is related with a Java class that was compiled with a higher version. That java version from which I built the image was `openjdk version "11.0.21" 2023-10-17`. Could you please check this out? Thanks!

**Comment by phuongnq** [comment]
@Euquimides I don't see the same error. The error you are seeing says that the file is built with Java 17, but the runtime (docker) is Java 11. Please make sure you have built 4.0.x with Java 11.
Here is what I have built and run:

1. In the craftercms, make sure Java is 11 and all modules in the branch `support/4.0.x` then run:
```
./gradlew selfupdate
./gradlew upgrade (make sure you also add the Social&Profile flags)
./gradlew bundle
```
2. In the docker-images repo, make sure Java is 11, the branch is `support/4.0.x` and build. You can also remove the folder `build` and `.gradle` to make sure it builds in a clean environment:
```
./gradlew build -PauthoringBundle=/home/pnguyen/craftercms/support_4.0.x/craftercms/bundles/crafter-cms-authoring-4.0.8-SNAPSHOT-linux-x86_64.tar.gz -PdeliveryBundle=/home/pnguyen/craftercms/support_4.0.x/craftercms/bundles/crafter-cms-delivery-4.0.8-SNAPSHOT-linux-x86_64.tar.gz  -Ptag=4.0.8-SNAPSHOT
```
3. In the docker-compose repo, make sure the branch is `support/4.0.x` and run:
```
cd scripts
./start.sh authoring
./logs.sh authoring
```
=> Notice that Authoring starts normally and you can login to Studio.

If this does not work for you, could you let me know the exact steps which caused the issue?

![2024-02-08_11-14](https://github.com/craftercms/craftercms/assets/2996543/7a1dba37-df3b-4216-b8cf-3b3924a2bd38)
![2024-02-08_11-13](https://github.com/craftercms/craftercms/assets/2996543/7992b18f-795c-4e23-b73a-baa141d65bef)

**Comment by Euquimides** [comment]
Hi @phuongnq ! Regarding this issue. I made sure to check the Java version when building Crafter, deleted the `build` folder and I was able to create the image and start Docker Compose without problems. Now, what I am noticing is the following both in 4.0.x and 4.1.x in CE and EE:

In the case of EE, when creating the image with Profile included, Docker Compose starts both Studio and Profile, so the Profile URL is available, but when login to Profile I get the following error:
```
authoring-profile-1   | 2024-02-20_20:32:46.284 [http-nio-8080-exec-5] WARN  org.craftercms.security.utils.tenant.AllTenantsResolver - Unable to retrieve tenants
authoring-profile-1   | org.craftercms.profile.api.exceptions.ProfileException: I/O error on GET request for "http://tomcat:8080/crafter-profile/api/1/tenant/all.json": Attempted read from closed stream.; nested exception is java.io.IOException: Attempted read from closed stream.
authoring-profile-1   | 	at org.craftercms.profile.services.impl.AbstractProfileRestClientBase.handleException(AbstractProfileRestClientBase.java:185) ~[crafter-profile-client-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.profile.services.impl.AbstractProfileRestClientBase.doGetForObject(AbstractProfileRestClientBase.java:150) ~[crafter-profile-client-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.profile.services.impl.TenantServiceRestClient.getAllTenants(TenantServiceRestClient.java:105) ~[crafter-profile-client-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.utils.tenant.TenantUtils.getTenantNames(TenantUtils.java:47) ~[crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.utils.tenant.AllTenantsResolver.getTenants(AllTenantsResolver.java:46) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.processors.impl.LoginProcessor.processRequest(LoginProcessor.java:135) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.processors.impl.RequestSecurityProcessorChainImpl.processRequest(RequestSecurityProcessorChainImpl.java:58) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.processors.impl.ReturnCurrentAuthenticationProcessor.processRequest(ReturnCurrentAuthenticationProcessor.java:75) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.processors.impl.RequestSecurityProcessorChainImpl.processRequest(RequestSecurityProcessorChainImpl.java:58) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.processors.impl.CurrentAuthenticationResolvingProcessor.processRequest(CurrentAuthenticationResolvingProcessor.java:85) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.processors.impl.RequestSecurityProcessorChainImpl.processRequest(RequestSecurityProcessorChainImpl.java:58) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.processors.impl.AddSecurityCookiesProcessor.processRequest(AddSecurityCookiesProcessor.java:73) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.processors.impl.RequestSecurityProcessorChainImpl.processRequest(RequestSecurityProcessorChainImpl.java:58) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.servlet.filters.RequestSecurityFilter.doFilterInternal(RequestSecurityFilter.java:138) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.security.servlet.filters.RequestSecurityFilter.doFilter(RequestSecurityFilter.java:109) [crafter-security-provider-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:354) [spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:267) [spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.craftercms.commons.http.RequestContextBindingFilter.doFilter(RequestContextBindingFilter.java:78) [crafter-commons-utilities-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) [spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117) [spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71) [log4j-web-2.20.0.jar:2.20.0]
authoring-profile-1   | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342) [catalina.jar:9.0.85]
authoring-profile-1   | 	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390) [tomcat-coyote.jar:9.0.85]
authoring-profile-1   | 	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) [tomcat-coyote.jar:9.0.85]
authoring-profile-1   | 	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928) [tomcat-coyote.jar:9.0.85]
authoring-profile-1   | 	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1794) [tomcat-coyote.jar:9.0.85]
authoring-profile-1   | 	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) [tomcat-coyote.jar:9.0.85]
authoring-profile-1   | 	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-util.jar:9.0.85]
authoring-profile-1   | 	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-util.jar:9.0.85]
authoring-profile-1   | 	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:9.0.85]
authoring-profile-1   | 	at java.lang.Thread.run(Thread.java:840) [?:?]
authoring-profile-1   | Caused by: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://tomcat:8080/crafter-profile/api/1/tenant/all.json": Attempted read from closed stream.; nested exception is java.io.IOException: Attempted read from closed stream.
authoring-profile-1   | 	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:791) ~[spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:717) ~[spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:637) ~[spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.craftercms.profile.services.impl.AbstractProfileRestClientBase.doGetForObject(AbstractProfileRestClientBase.java:146) ~[crafter-profile-client-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	... 44 more
authoring-profile-1   | Caused by: java.io.IOException: Attempted read from closed stream.
authoring-profile-1   | 	at org.apache.http.impl.io.ChunkedInputStream.read(ChunkedInputStream.java:176) ~[httpcore-4.4.16.jar:4.4.16]
authoring-profile-1   | 	at org.apache.http.conn.EofSensorInputStream.read(EofSensorInputStream.java:135) ~[httpclient-4.5.14.jar:4.5.14]
authoring-profile-1   | 	at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:270) ~[?:?]
authoring-profile-1   | 	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:313) ~[?:?]
authoring-profile-1   | 	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:188) ~[?:?]
authoring-profile-1   | 	at java.io.InputStreamReader.read(InputStreamReader.java:177) ~[?:?]
authoring-profile-1   | 	at java.io.Reader.read(Reader.java:250) ~[?:?]
authoring-profile-1   | 	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1612) ~[commons-io-2.14.0.jar:2.14.0]
authoring-profile-1   | 	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1591) ~[commons-io-2.14.0.jar:2.14.0]
authoring-profile-1   | 	at org.apache.commons.io.IOUtils.copy(IOUtils.java:1386) ~[commons-io-2.14.0.jar:2.14.0]
authoring-profile-1   | 	at org.apache.commons.io.IOUtils.copy(IOUtils.java:1155) ~[commons-io-2.14.0.jar:2.14.0]
authoring-profile-1   | 	at org.apache.commons.io.IOUtils.toString(IOUtils.java:3167) ~[commons-io-2.14.0.jar:2.14.0]
authoring-profile-1   | 	at org.craftercms.commons.rest.HttpMessageConvertingResponseErrorHandler.getResponseBodyAsString(HttpMessageConvertingResponseErrorHandler.java:88) ~[crafter-commons-utilities-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.craftercms.commons.rest.HttpMessageConvertingResponseErrorHandler.handleError(HttpMessageConvertingResponseErrorHandler.java:76) ~[crafter-commons-utilities-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63) ~[spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:825) ~[spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:783) ~[spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:717) ~[spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:637) ~[spring-web-5.3.31.jar:5.3.31]
authoring-profile-1   | 	at org.craftercms.profile.services.impl.AbstractProfileRestClientBase.doGetForObject(AbstractProfileRestClientBase.java:146) ~[crafter-profile-client-4.1.3-SNAPSHOT.jar:4.1.3-SNAPSHOT]
authoring-profile-1   | 	... 44 more
```
![image](https://github.com/craftercms/craftercms/assets/90878288/d8538ff0-8581-41ae-8047-b3389232482a)

Checking lines up, when Docker Compose starts building, I noticed the following:
```
authoring-profile-1   | 2024-02-20_19:58:47.913 [main] WARN  org.springframework.web.context.support.XmlWebApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'crafter.tenantService' defined in class path resource [crafter/profile/services-context.xml]: Cannot resolve reference to bean 'crafter.profileService' while setting bean property 'profileService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'crafter.profileService' defined in class path resource [crafter/profile/services-context.xml]: Cannot resolve reference to bean 'crafter.entitlementValidator' while setting bean property 'entitlementValidator'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'crafter.entitlementValidator' defined in class path resource [crafter/entitlements/entitlements-context.xml]: Invocation of init method failed; nested exception is org.craftercms.commons.entitlements.exception.InvalidLicenseException: License found but could not be loaded
```
Seems like Studio is reading the license correctly but Profile isn't reading it?

In the case of CE is different, there is no message in the logs when trying to access Profile. I've made sure that Profile and Social are indeed working in my 4.0.x CE build for example, but when trying to access the URL `localhost:port/crafter-profile-admin` I get the following message:
![image](https://github.com/craftercms/craftercms/assets/90878288/ad493635-1b30-4296-b4ef-8ec68b4eb6ce)

Could you please take a look at this? Thanks a lot for your help!

**Comment by phuongnq** [comment]
Hi @Euquimides I was not able to reproduce the issue.
Here is my yaml file:

```
version: '3.7'
services:
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:7.17.1
    ports:
      - 9201:9200
    environment:
      - discovery.type=single-node
      - bootstrap.memory_lock=true
      - "ES_JAVA_OPTS=-Xss1024K -Xmx1G"
    ulimits:
      memlock:
        soft: -1
        hard: -1
    volumes:
      - elasticsearch_data:/usr/share/elasticsearch/data
      - elasticsearch_logs:/usr/share/elasticsearch/logs
  tomcat:
    image: craftercms/authoring_tomcat:4.0.8-SNAPSHOT # craftercms version flag
    depends_on:
      - elasticsearch
      - deployer
    ports:
      - 8080:8080
    volumes:
      - crafter_data:/opt/crafter/data
      - crafter_logs:/opt/crafter/logs
      - crafter_temp:/opt/crafter/temp
      # Elastic Search dirs needed for backup/restore
      - elasticsearch_data:/opt/crafter/data/indexes-es
    environment:
      - DEPLOYER_HOST=deployer
      - DEPLOYER_PORT=9191
      - ES_HOST=elasticsearch
      - ES_PORT=9200
  deployer:
    image: craftercms/deployer:4.0.8-SNAPSHOT # craftercms version flag
    depends_on:
      - elasticsearch
    ports:
      - 9191:9191
    volumes:
      - crafter_data:/opt/crafter/data
      - crafter_logs:/opt/crafter/logs
      - crafter_temp:/opt/crafter/temp
    environment:
      - TOMCAT_HOST=tomcat
      - TOMCAT_HTTP_PORT=8080
      - ES_HOST=elasticsearch
      - ES_PORT=9200
  mongodb:
    image: mongo:4.0.28
    ports:
      - 27017:27017
    volumes:
      - mongo_data:/data/db
      - mongo_logs:/var/log/mongodb
    environment:
      MONGO_INITDB_DATABASE: crafterprofile
  profile:
    image: craftercms/profile:4.0.8-SNAPSHOT
    depends_on:
      - mongodb
    ports:
      - 8081:8080
    volumes:
      - crafter_data:/opt/crafter/data
      - crafter_logs:/opt/crafter/logs
      - crafter_temp:/opt/crafter/temp
      - mongo_data:/opt/crafter/data/mongodb
      - mongo_logs:/opt/crafter/logs/mongodb
    environment:
      - MONGODB_HOST=mongodb
      - MONGODB_PORT=27017
volumes:
  elasticsearch_data:
    name: crafter_authoring_data_elasticsearch
  elasticsearch_logs:
    name: crafter_authoring_logs_elasticsearch
  crafter_data:
    name: crafter_authoring_data
  crafter_logs:
    name: crafter_authoring_logs
  crafter_temp:
    name: crafter_authoring_temp
  mongo_data:
    name: crafter_authoring_data_mongo
  mongo_logs:
    name: crafter_authoring_logs_mongo
```

And enterprise yaml file:
```
version: '3.7'
services:
  tomcat:
    image: craftercms/authoring_tomcat:4.0.8E # craftercms enterprise version flag
    volumes:
      # License (copy your license under ./license/crafter.lic)
      - ./license/crafter.lic/:/opt/crafter/bin/apache-tomcat/shared/classes/crafter/license/crafter.lic
  deployer:
    image: craftercms/deployer:4.0.8E # craftercms enterprise version flag
    volumes:
      # License (copy your license under ./license/crafter.lic)
      - ./license/crafter.lic/:/opt/crafter/bin/apache-tomcat/shared/classes/crafter/license/crafter.lic
  profile:
    image: craftercms/profile:4.0.8E # craftercms enterprise version flag
    volumes:
      # License (copy your license under ./license/crafter.lic)
      - ./license/crafter.lic/:/opt/crafter/bin/apache-tomcat/shared/classes/crafter/license/crafter.lic
```
By any chance you have a different Mongo version? In 4.0.x, we should use Mongo with version 4.0.28

**Comment by Euquimides** [comment]
Hi @phuongnq ! Indeed it was the version I was adding to the yaml file, once I added the correct version it worked correctly. 

Fix verified for Crafter v4.0.8 CE and EE. Thank you! Closing this ticket.
